tests/ui/static/addr-of-static-mut.rs

```diff
@@ -0,0 +1,10 @@
+//@ check-pass
+
+// see https://github.com/rust-lang/rust/issues/125833
+// notionally, taking the address of a static mut is a safe operation,
+// as we only point at it instead of generating a true reference to it
+static mut FLAG: bool = false;
+fn main() {
+    let p = std::ptr::addr_of!(FLAG);
```

Comment:
The `mut-ptr-to-static.rs`  test seems to imply that `addr_of_mut!` should be tested the same way. Also the PR title / commit/merge message should make it clearer how `addr_of_mut!` is affected or not.

Reply:
@briansmith I will make that clearer, but fwiw `addr_of_mut!` will be operationally identical when considering a `static mut`.